tests/Doctrine/Tests/ORM/Functional/Ticket/GH10348Test.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class GH10348Test extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10348ChildEntity::class,
+            GH10348ParentEntity::class,
+        ]);
+    }
+
+    public function testCanRemoveParentWithChildRelatesToOwnEntity(): void
```

Comment:
I think it would be best to use the same vocabulary as the ORM. This is a self-referencing entity.
```suggestion
    public function testCanRemoveParentWithSelfReferencingChild(): void
```